english/tests/tests.py

```diff
@@ -5,121 +5,492 @@
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data), 3)
         self.assertDictEqual(dict(response.data[0]), expected_data)
 
-    def test_get_classes_for_current_student(self):
+    def test_get_classes_success(self):
+        self.client.force_login(user=self.student)
         expected_data = {
             "pk": 1,
             "name": "FirstClass",
             "students": [
-                {"pk": 2, "first_name": "Matthew", "last_name": "Becher"},
                 {"pk": 3, "first_name": "Ayden", "last_name": "Henris"},
-                {"pk": 4, "first_name": "Brody", "last_name": "Nelson"},
-                {"pk": 1, "first_name": "Maksim", "last_name": "Lukash"},
+                {"pk": 4, "first_name": "Damian", "last_name": "Ward"},
+                {"pk": 5, "first_name": "Jonah", "last_name": "Matthews"},
+                {"pk": 1, "first_name": "Maksim", "last_name": "Lukash"}
             ],
-            "teacher": {"pk": 5, "first_name": "Damian", "last_name": "Ward"},
+            "teacher": {"pk": 2, "first_name": "Matthew", "last_name": "Becher"},
+            "days": ["Monday", "Wednesday", "Friday"],
+            "start_time": "18:00:00",
+            "end_time": "19:30:00"
         }
-        url = reverse("student-classes")
+        url = reverse("student-classes-list")
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(len(response.data), 2)
+        self.assertEqual(len(response.data), 3)
         self.assertDictEqual(dict(response.data[0]), expected_data)
 
-    def test_get_classes_for_current_teacher(self):
+    def test_get_classes_failed(self):
+        self.client.force_login(user=self.teacher)
+        url = reverse("student-classes-list")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_leave_class(self):
+        data = {"pk": 1, "first_name": "Maksim", "last_name": "Lukash"}
+
+        self.client.force_login(user=self.teacher)
+        url = reverse("teacher-classes-retrive-update-destroy", kwargs={"pk": 1})
+        response = self.client.get(url)
+        self.assertIn(data, response.data["students"])
+
+        self.client.force_login(user=self.student)
+        url = reverse("join-to-class", kwargs={"pk": 1})
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+        self.client.force_login(user=self.teacher)
+        url = reverse("teacher-classes-retrive-update-destroy", kwargs={"pk": 1})
+        response = self.client.get(url)
+        self.assertNotIn(data, response.data["students"])
+
+    def test_join_class(self):
+        self.client.force_login(user=self.student)
+        data = {"pk": 1, "first_name": "Maksim", "last_name": "Lukash"}
+
+        url = reverse("join-to-class", kwargs={"pk": 1})
```

Comment:
Remove it without request. And check that this user is not in the student list using DB